site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/EditOAuth2AppPage.tsx

```diff
@@ -0,0 +1,131 @@
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import {
+  deleteOAuth2App,
+  deleteOAuth2AppSecret,
+  postOAuth2AppSecret,
+  putOAuth2App,
+} from "api/api";
+import type * as TypesGen from "api/typesGenerated";
+import {
+  oauth2App,
+  oauth2AppSecrets,
+  oauth2AppSecretsKey,
+} from "api/queries/oauth2";
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils";
+import { FC, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { EditOAuth2AppPageView } from "./EditOAuth2AppPageView";
+import { pageTitle } from "utils/page";
+import { Helmet } from "react-helmet-async";
+
+const EditOAuth2AppPage: FC = () => {
+  const navigate = useNavigate();
+  const { appId } = useParams() as { appId: string };
+  const queryClient = useQueryClient();
+  const [newAppSecret, setNewAppSecret] = useState<
+    TypesGen.OAuth2AppSecretFull | undefined
+  >(undefined);
```

Comment:
Oh weird.  I got an error when doing `SetNewAppSecret(undefined)` so I added that.  Could be I was just confused though.

This secret is different from its version in the list of secrets because it has the full secret (you only get to see the full secret once, when the secret is first cretaed).  The list only has truncated secrets, so a cross-reference would not work unfortunately.

Reply:
Oh is it because I did `useState(undefined)` instead of `useState()`?  I will try that.

Reply:
I should call them `fullSecret` and `truncatedSecrets` or something like that, because it is definitely not obvious.